lib/manager.dart

```diff
@@ -33,7 +33,7 @@ class SocketIOManager {
     MethodChannelNames.streamsChannel,
   );
 
-  final _sockets = <int, SocketIO>{};
+  final _sockets = <int?, SocketIO>{};
```

Comment:
the key is supposed to be strict `int`. Any other possibilities are to be filtered out in usages below.

Reply:
The problem is that: -
    final index = await _channel.invokeMethod<int>(
      PlatformMethod.newInstance,
      {
        'options': options.asMap(),
        'clear': _clearExisting,
      },
    );

Returns a Future<int?> as per the MethodChannel contract so we would need to provide a default with ??.